stats/members/mdetail.php

```diff
@@ -9,12 +9,20 @@
 include_once('../includes/team.inc');
 include_once('../includes/member.inc');
 
-$tally_name = array_get( $_GET, 'tally_name', null );
-
 $id = get_integer_param($_GET, 'id', null, 0, null);
-
 $user = User::load_from_uid($id);
 
+$valid_tally_names = array_keys($page_tally_names);
+try
+{
+    $tally_name = get_enumerated_param($_GET, 'tally_name', null, $valid_tally_names, true);
+}
+catch(InvalidArgumentException $exception)
+{
+    die(sprintf(_("tally_name must be one of: %s"), implode(", ", $valid_tally_names)));
```

Comment:
I thought about that but decided that in some instance we might not want to inform the user what the valid options were, because we may not want them futzing with the URL or POST data directly. It's conceivable doing so might open up the ability for a malicious party to type malformed data to fish for valid data.

Reply:
Hm, possible. In most cases, the malicious party could figure out the valid options anyway, just by looking at the published code. But there's at least one case (in tasks.php, with 'task_assignee') where the enumeration is constructed dynamically, and the site might not want to reveal it. (Any others?)

My first inclination would be to make the verbose message the "default" and have the few places that need to hide the enumeration values construct a non-revealing message instead. Or I suppose we could define a subclass of InvalidArgumentException that gives a choice of 2 messages, one verbose and one not, so the caller doesn't have to construct anything, it just needs to pick which it wants.

Reply:
After thinking about it some I can't see that we have many places that we care if the user knows about the valid options so I've coded it up so that we always list the choices in the exception.